src/radical/pilot/agent/launch_method/srun.py

```diff
@@ -129,15 +129,24 @@ def get_launch_cmds(self, task, exec_path):
             # the scheduler did place tasks - we can't honor the core and gpu
             # mapping (see above), but we at least honor the nodelist.
             nodelist = [rank['node_name'] for rank in slots['ranks']]
+            n_nodes  = len(set(nodelist))
 
             # older slurm versions don't accept nodefiles
+            # 42 node is the upper limit to switch from `--nodelist`
+            # to `--nodefile`
             if self._vmajor > 18:
-                nodefile = '%s/%s.nodes' % (sbox, uid)
-                with ru.ru_open(nodefile, 'w') as fout:
-                    fout.write(','.join(nodelist))
-                    fout.write('\n')
+                if n_nodes > 42:
```

Comment:
```python
if node > 42:

    if self._vmajor <= 18:
        raise Exception(...)

    nodefile = ...
```

btw I also would propose to move these numbers up, as constants (names for constants just an example, you can come up with yours)

```python
MIN_NUM_NODES_IN_NODELIST = 42
MIN_SLURM_VER_TO_NODELIST = 18
```

Reply:
@mtitov  This looks good to set the min version of `slurm` and `number of nodes`